shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1396,12 +1504,28 @@ - (void)setupKeyboardAnimationVsyncClient {
       // Ensure the keyboardAnimationView is in view hierarchy when animation running.
       [flutterViewController.get().view addSubview:[flutterViewController keyboardAnimationView]];
     }
-    if ([flutterViewController keyboardAnimationView].layer.presentationLayer) {
-      CGFloat value =
-          [flutterViewController keyboardAnimationView].layer.presentationLayer.frame.origin.y;
-      flutterViewController.get()->_viewportMetrics.physical_view_inset_bottom = value;
-      [flutterViewController updateViewportMetrics];
-    }
+
+    KeyboardAnimationView* keyboardView =
+        (KeyboardAnimationView*)[flutterViewController keyboardAnimationView];
+
+    fml::TimeDelta timeElapsed = recorder.get()->GetVsyncTargetTime() - keyboardView.startTime;
+    double frameRate = [[flutterViewController keyboardAnimationVSyncClient] getRefreshRate];
+
+    double animationSettlingDuration = [flutterViewController keyboardSpringCurve].settlingDuration;
+    double keyboardAnimationTimeMillis = animationSettlingDuration * 1000;
+
+    double expectedFrames = frameRate * animationSettlingDuration;
+    double percentComplete = timeElapsed.ToMillisecondsF() / keyboardAnimationTimeMillis;
+
+    int frameApproximation = roundf(MIN(percentComplete, 1) * expectedFrames);
+    double animationFramePercentage = [[[flutterViewController keyboardAnimationStops]
+        objectAtIndex:frameApproximation] doubleValue];
```

Comment:
Maybe add a safe check for `frameApproximation`, as it will be the index of `keyboardAnimationStops`, we should make sure that it is in range or it will not be larger than `keyboardAnimationStops.count - 1`, or this will cause crash.

Reply:
agree, I had it in before but deleted :)